Notes-CaaSP.md

```diff
@@ -132,11 +132,177 @@ kubectl create secret generic ceph-secret-admin
+
+* Create a openstack project to run CaaSP in (e.g. caasp), add a user as admin and export the project to be used by terraform
+
+  ```
+  openstack project create --domain default --description "CaaSP Project" caasp
+  openstack role add --project caasp --user admin admin
+  export OS_PROJECT_NAME='caasp'
+  ```
+
+* Create a Openstack network plus a subnet for caasp (e.g. caasp-net) and add a router to the extrenal (e.g. floating) network
+
+  ```
+  openstack network create caasp-net
+  openstack subnet create caasp_subnet --network caasp-net --subnet-range 10.0.2.0/24
+  openstack router create caasp-net-router
+  openstack router set caasp-net-router --external-gateway floating
+  openstack router add subnet caasp-net-router caasp_subnet
+  ```
+
+**Mandatory Steps**
+
+The following steps have to be done at least once in order to be able to deploy CaaSP and CAP on top of OpenStack
+
+
+* Clone the terraform script
+
+  ```
+  git clone git@github.com:kubic-project/automation.git
+  cd automation/caasp-openstack-terraform
+  ```
+
+* Upload the [CaaSPv2 image](http://dist.suse.de/install/SUSE-CaaSP-2.0-GM/SUSE-CaaS-Platform-2.0-for-OpenStack-Cloud.x86_64-2.0.0-GM.qcow2)
+
+  ```
+  wget http://dist.suse.de/install/SUSE-CaaSP-2.0-GM/SUSE-CaaS-Platform-2.0-for-OpenStack-Cloud.x86_64-2.0.0-GM.qcow2
+  openstack image create --file SUSE-CaaS-Platform-2.0-for-OpenStack-Cloud.x86_64-2.0.0-GM.qcow2 SUSE-CaaS-Platform-2.0-GM
+  ```
+
+* Create a additional security group with rules needed for CAP
+
+  ```
+  openstack security group create cap --description "Allow CAP traffic"
+  openstack security group rule create cap --protocol any --dst-port any --ethertype IPv4 --egress
+  openstack security group rule create cap --protocol any --dst-port any --ethertype IPv6 --egress
+  openstack security group rule create cap --protocol tcp --dst-port 20000:20008 --remote-ip 0.0.0.0/0
+  openstack security group rule create cap --protocol tcp --dst-port 443:443 --remote-ip 0.0.0.0/0
+  openstack security group rule create cap --protocol tcp --dst-port 2793:2793 --remote-ip 0.0.0.0/0
+  openstack security group rule create cap --protocol tcp --dst-port 4443:4443 --remote-ip 0.0.0.0/0
+  openstack security group rule create cap --protocol tcp --dst-port 80:80 --remote-ip 0.0.0.0/0
+  openstack security group rule create cap --protocol tcp --dst-port 2222:2222 --remote-ip 0.0.0.0/0
+  ```
+
+* Edit `openstack.tfvars`. Use the names of the just created OpenStack objects
+
+  Example:
+  ```
+  image_name = "SUSE-CaaS-Platform-2.0-GM"
+  internal_net = "caasp-net"
+  external_net = "floating"
+  admin_size = "m1.large"
+  master_size = "m1.large"
+  masters = 1
+  worker_size = "m1.xlarge"
+  workers = 2
+  ```
+
+* Initialize terraform
+
+  ```
+  terraform init
+  ```
+
+
+### Deploy CaaSP
+
+* Source the openrc.sh file, set the project and deploy
+
+  ```
+  . openrc.sh
```

Comment:
This already was done at the beginning after downloading the file.

Reply:
yes, but the steps above are only done once. Successive deployments will skip those and then it is not sourced.